fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/ThenQueryKeyExpression.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * ThenQueryKeyExpression.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.expressions;
+
+import com.apple.foundationdb.record.RecordMetaDataProto;
+import com.apple.foundationdb.record.metadata.Key;
+import com.apple.foundationdb.record.metadata.expressions.KeyExpression;
+import com.apple.foundationdb.record.metadata.expressions.QueryableKeyExpression;
+import com.apple.foundationdb.record.metadata.expressions.ThenKeyExpression;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecord;
+import com.apple.foundationdb.record.query.plan.cascades.CorrelationIdentifier;
+import com.apple.foundationdb.record.query.plan.cascades.values.Value;
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Message;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * A simple {@link QueryableKeyExpression} that covers a {@link ThenKeyExpression}, translating
+ * into a Tuple for comparison operations with {@link QueryKeyExpressionWithComparison}.
+ */
+public class ThenQueryKeyExpression implements QueryableKeyExpression {
```

Comment:
This approach obligates the planner to deal with `WHERE (a,b) = ?`, as opposed to `WHERE (a,b) = (?,?)` or `WHERE a = ? AND b = ?`. Today it only really handles the last one.

A `QueryableKeyExpression` needs to match an _element_ of the index's root expression. And this will likely be a subsequence.

An alternative approach that is perhaps more expedient is to treat this as two joins, either in `addJoin` right away or in `JoinedRecordPlanner`.